src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -9999,7 +9999,61 @@ def describe(
         # max            NaN      3.0    NaN
         sorted_percentiles = sorted(percentiles)
         dtypes = self.dtypes
-        query_compiler = self
+        # If we operate on the original frame's labels, then if two columns have the same name but
+        # different dtypes, the JOIN behavior of SnowflakeQueryCompiler.concat will produce incorrect
```

Comment:
I want to make sure my understanding is correct, it is not actual an problem when the both columns have different dtypes, it is when one is numeric, and another is non-numeric, because the applicable function on the columns are different, one is top/freq, another one is min/max etc. If there are both numeric, but just different type, i imaging it should be ok, because the min, max df will have both columns in the query compiler, and concat should be able to take care of it correctly.

Reply:
Yes, your understanding is correct. We have test cases verifying this as well.

Reply:
can you update the comment then, it is kind if misleading